### PR TITLE
Allow to use string to define StrokeShape in XAML

### LINKED ISSUE #2690: [Bug] Border StrokeShape needs typeconverter for XAML
### Description

This works

```
<Border Stroke="Red"
                        StrokeThickness="3"
                        HorizontalOptions="Start"
                        Padding="12,8"
                        >
                    <Border.StrokeShape>
                        <Rectangle/>
                    </Border.StrokeShape>
                    <Label Text="Images" TextColor="#999999" 
                           FontSize="42"
                           Background="Green"
                           HorizontalOptions="Start"
                           StyleClass="Headline"/>
                </Border>
```

This doesn't

```
<Border Stroke="Red"
                        StrokeThickness="3"
                        StrokeShape="Rectangle"
                        HorizontalOptions="Start"
                        Padding="12,8"
                        >
                    <Label Text="Images" TextColor="#999999" 
                           FontSize="42"
                           Background="Green"
                           HorizontalOptions="Start"
                           StyleClass="Headline"/>
                </Border>
```

image

### Basic Information

- Version with issue: pre9

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Added TypeConverter (and tests) to allow set the StrokeShape like:

- fixes #2690 

```
StrokeShape="Rectangle"
StrokeShape="Ellipse"
StrokeShape="Polygon 0 48, 0 144, 96 150, 100 0, 192 0, 192 96, 50 96, 48 192, 150 200 144 48"
StrokeShake="RoundRectangle 20, 0, 20, 0"
```

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)

#### Does this PR touch anything that might affect accessibility?
- No

### CODE REVIEW

**Comment by hartez** [comment]
This probably needs a test in the XAML unit tests, so we can ensure it works with/without XAMLC.

**Review by rmarinho — CHANGES_REQUESTED** [review]
Can you had a xaml unit test ? with xamlc

**Comment by jsuarezruiz** [comment]
Sure, adding XAML tests.

**Review by StephaneDelcroix — APPROVED** [review]

**Comment by rmarinho** [comment]
@jsuarezruiz unit tests are failing

**Comment by jsuarezruiz** [comment]
> @jsuarezruiz unit tests are failing

I have checked them locally and are passing. I've done rebase and I'll be waiting for the Build. 

image

File: src/Controls/src/Core/Shapes/StrokeShapeTypeConverter.cs [hunk 1]
```diff
@@ -1,35 +1,114 @@
...
 using System.ComponentModel;
 using System.Globalization;
+using Microsoft.Maui.Converters;
+using Microsoft.Maui.Graphics;
 
 namespace Microsoft.Maui.Controls.Shapes
 {
 	public class StrokeShapeTypeConverter : TypeConverter
 	{
+		public const string Ellipse = "ellipse";
```
Comment (StephaneDelcroix) [inline src/Controls/src/Core/Shapes/StrokeShapeTypeConverter.cs:11]: use `nameof()`

File: src/Controls/tests/Xaml.UnitTests/Border.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		}
+
+		[TestFixture]
+		class Tests
+		{
+			[SetUp] public void Setup() => Device.PlatformServices = new MockPlatformServices();
+			[TearDown] public void TearDown() => Device.PlatformServices = null;
+
+			[Test]
+			public void InitializeStrokeShape(bool useCompiledXaml)
```
Comment (StephaneDelcroix) [inline src/Controls/tests/Xaml.UnitTests/Border.xaml.cs:22]:
```suggestion
			public void InitializeStrokeShape([Values (false, true)]bool useCompiledXaml)
```

**Review by rmarinho — APPROVED** [review]

**Pull request merged**